src/consumer/itinerary-item/poll.ts

```diff
@@ -163,15 +164,24 @@ export class Poll implements IEntity, IDestroyable {
             this.listenForQuestions()
           }
           this.listenForResponses()
+          this.socket.removeListener(correlationId, callback)
```

Comment:
Can delete this as done in finally
```suggestion
```

Reply:
keeping in because the callback is triggered inside a separate event listener separate to the initial promise wrapper where finally runs.
the new event listener is created after finally will run.
added a `isReconnecting` check to only remove eventlistener inside callback if it is a reconnection